toolkit/tools/pkg/imagecustomizerlib/configvalidation.go

```diff
@@ -503,14 +517,44 @@ func validateIsoPxeCustomization(rc *ResolvedConfig) error {
 		// While defining a storage configuration can work when the input image is
 		// an iso, there is no obvious point of moving content between partitions
 		// where all partitions get collapsed into the squashfs at the end.
-		if rc.Config.CustomizePartitions() {
+		if len(rc.Storage.Disks) > 0 {
 			return ErrCannotCustomizePartitionsOnIso
 		}
 	}
 
 	return nil
 }
 
+func validateStorage(configChain []*ConfigWithBasePath) error {
+	var baseHasDisks bool
+
+	for i, configWithBase := range configChain {
+		storage := configWithBase.Config.Storage
+
+		hasDisks := len(storage.Disks) > 0
+		hasResetUUID := storage.ResetPartitionsUuidsType != imagecustomizerapi.ResetPartitionsUuidsTypeDefault
+		hasReinitVerity := storage.ReinitializeVerity != imagecustomizerapi.ReinitializeVerityTypeDefault
+
+		if baseHasDisks {
+			if hasResetUUID {
+				return fmt.Errorf(
+					"cannot specify 'resetPartitionsUuidsType' in config at layer %d when a base config specifies '.storage.disks'", i)
```

Comment:
I don't think `at layer %d` is particularly meaningful to the user, since they see the config tree and not the list. If possible, it might be nice to give the config file names. If not, I would just remove `at layer %d`.